sormas-backend/src/main/java/de/symeda/sormas/backend/vaccination/VaccinationFacadeEjb.java

```diff
@@ -43,28 +61,151 @@
 		VaccinationDto existingDto = toDto(existingVaccination);
+		Date currentVaccinationDate = existingVaccination != null ? existingVaccination.getVaccinationDate() : null;
 
 		Pseudonymizer pseudonymizer = Pseudonymizer.getDefault(userService::hasRight);
 		restorePseudonymizedDto(dto, existingDto, existingVaccination, pseudonymizer);
 
-		validate(dto);
+		validate(dto, false);
 
 		existingVaccination = fillOrBuildEntity(dto, existingVaccination, true);
 		vaccinationService.ensurePersisted(existingVaccination);
 
+		updateVaccinationStatuses(
+			existingVaccination.getVaccinationDate(),
+			currentVaccinationDate,
+			existingVaccination.getImmunization().getPerson().getId(),
+			existingVaccination.getImmunization().getDisease());
+
 		return convertToDto(existingVaccination, pseudonymizer);
 	}
 
+	@Override
+	public VaccinationDto create(
+		VaccinationDto dto,
+		RegionReferenceDto region,
+		DistrictReferenceDto district,
+		PersonReferenceDto person,
+		Disease disease) {
+
+		if (dto.getUuid() != null && vaccinationService.getByUuid(dto.getUuid()) != null) {
+			throw new IllegalArgumentException("DTO already has a UUID");
+		}
+
+		validate(dto, true);
+
+		Vaccination vaccination = null;
+		vaccination = fillOrBuildEntity(dto, vaccination, true);
+		boolean immunizationFound = addImmunizationToVaccination(vaccination, person.getUuid(), disease);
+
+		if (!immunizationFound) {
+			ImmunizationDto immunizationDto = ImmunizationDto.build(person);
+			immunizationDto.setDisease(disease);
+			immunizationDto.setResponsibleRegion(region);
+			immunizationDto.setResponsibleDistrict(district);
+			immunizationDto.setReportingUser(userService.getCurrentUser().toReference());
+			immunizationDto.setMeansOfImmunization(MeansOfImmunization.VACCINATION);
+			immunizationFacade.save(immunizationDto);
+
+			Immunization immunization = immunizationService.getByUuid(immunizationDto.getUuid());
+			vaccination.setImmunization(immunization);
+		}
+
+		if (vaccination.getImmunization() == null) {
+			throw new ValidationRuntimeException(I18nProperties.getValidationError(Validations.validImmunization));
+		}
+
+		vaccinationService.ensurePersisted(vaccination);
+
+		updateVaccinationStatuses(vaccination.getVaccinationDate(), null, vaccination.getImmunization().getPerson().getId(), disease);
+
+		return convertToDto(vaccination, Pseudonymizer.getDefault(userService::hasRight));
+	}
+
+	private boolean addImmunizationToVaccination(Vaccination vaccination, String personUuid, Disease disease) {
+
+		List<Immunization> immunizations = immunizationService.getByPersonAndDisease(personUuid, disease);
+
+		if (immunizations.isEmpty()) {
+			return false;
+		}
+
+		if (immunizations.size() == 1) {
+			vaccination.setImmunization(immunizations.get(0));
+			return true;
+		}
+
+		// If the vaccination date is empty, add the vaccination to the latest immunization
```

Comment:
Suggestion: Add something like "Case 1: ..." to these 4 comments and then make sure that every case is covered in the unit test (add comments there as-well).